staging/src/k8s.io/kubectl/pkg/cmd/delete/delete_flags.go

```diff
@@ -155,6 +159,9 @@ func (f *DeleteFlags) AddFlags(cmd *cobra.Command) {
 	if f.Raw != nil {
 		cmd.Flags().StringVar(f.Raw, "raw", *f.Raw, "Raw URI to DELETE to the server.  Uses the transport specified by the kubeconfig file.")
 	}
+	if f.Interactive != nil {
+		cmd.Flags().BoolVarP(f.Interactive, "interactive", "i", *f.Interactive, "If true, delete resource only when user confirms. This flag is in Alpha.")
```

Comment:
I'm finding myself again wondering why this would be an exception to the alpha env var process we usually require. This is still an alpha as the comment says, and our stance has typically been that a doc string is not a sufficient indicator of a user's active consent to use experimental features. What is our justification in this case?

Reply:
I mimicked this usage https://github.com/kubernetes/kubernetes/blob/6c549d75a8d951ec43ecd5394b2634c1a40e5dd1/staging/src/k8s.io/kubectl/pkg/cmd/util/helpers.go#L503

I saw similar comment to yours by @soltysh above;

```
Lastly, should we hide this behind a env var until we gain confidence it's good enough, or at least we should mark the flag as alpha, to give a clean signal to users that we might change it in the next release.
```

I'm fine hiding this flag behind env var, if this is the correct approach.

Reply:
@ardaguclu I had a lengthy conversation with @KnVerey over the past few days and we concluded that it's best to write a KEP laying out all the possible approaches we've discussed during [sig-cli call last week](https://docs.google.com/document/d/1r0YElcXt6G5mOWxwZiXgGu_X6he3F--wKwg-9UBc29I/edit#heading=h.qtxa9691nj0t), as well as taking into account information from past efforts (https://github.com/kubernetes/enhancements/issues/2775 and https://github.com/kubernetes/enhancements/pull/2777), and the [lengthy discussion that happened on k-dev ](https://groups.google.com/g/kubernetes-dev/c/y4Q20V3dyOk). Especially, that this will be the first installment of interactivity mechanism in kubectl, so this will lay the ground for eventual future work around the topic. 

Lastly, given all the various discussions, I'd suggest writing the KEP asap and sharing it with k-dev and sig-cli mailing list, to gather more feedback about its functionality. This will also allow us more time for experiments, when we hide the flag behind feature gate (aka env var).

/hold 
for KEP, and 1.28

Reply:
Slightly related: I can imagine that we one day support setting a default for `interactive`. For example, a [`kuberc`](https://github.com/kubernetes/enhancements/tree/35befff0ad33187b2c93141d5fe1513a1b4a39a1/keps/sig-cli/3104-introduce-kuberc) setting to turn interactive on by default, on a _subcommand by subcommand_ basis.

&hellip; but also a way for scripts to set `KUBECTL_INTERACTIVE=never`, and a long (eg 5 releases) adoption period before script authors need to even think about the possible interactions here.

**No action required.**

Could be something to think about if / when we come to name environment variables.